crates/circuit/src/dag_circuit.rs

```diff
@@ -1556,7 +1556,14 @@ def _format(operand):
     ///     DAGCircuit: An empty copy of self.
     #[pyo3(signature = (*, vars_mode="alike"))]
     fn copy_empty_like(&self, py: Python, vars_mode: &str) -> PyResult<Self> {
-        let mut target_dag = DAGCircuit::new(py)?;
+        let mut target_dag = DAGCircuit::with_capacity(
+            py,
+            self.num_qubits(),
+            self.num_clbits(),
+            Some(self.num_vars()),
+            None,
```

Comment:
Shouldn't this be:

```suggestion
            None,
            Some(self.num_vars()),
```

Otherwise we potentially will be overallocating if there are vars but no ops. Although it might make more sense to flip the arguments in the `with_capacity` signature as doing `qubits, clbits, vars, ops` makes more sense intuitively to me.

Reply:
I'll re-arrange the function signature for this as it makes more sense, thank you 🚀